src/lib/viz/layer/Layer.ts

```diff
@@ -21,6 +22,16 @@ import { FunctionFilterApplicator } from '../filters/FunctionFilterApplicator';
 import { ColumnFilters } from '../filters/types';
 import { basicStyle } from '../style/helpers/basic-style';
 
+const DATA_READY_EVENT = 'layer:data:ready';
+const DATA_CHANGED_EVENT = 'layer:data:changed';
+
+export interface LayerDataState {
+  isFirstTime: boolean;
```

Comment:
isFirstTime corresponds to 'no-data-yet' ?

Reply:
the idea behind it is: "no events send yet" or being more specific, "ready event not sent yet". I can change it to `isReady` if it is more clear

Reply:
Changed to use 1 unique state, so this one does not applies